src/unix/stream.c

```diff
@@ -795,7 +799,7 @@ static void uv__write(uv_stream_t* stream) {
     msg.msg_iovlen = iovcnt;
     msg.msg_flags = 0;
 
-    msg.msg_control = (void*) scratch;
+    msg.msg_control = (void*) &scratch.alias;
```

Comment:
Good point, I overlooked it's being accessed as a cmsghdr further down.  That last definition looks correctest to me.